docs/internal/testing/testing.md

```diff
@@ -0,0 +1,64 @@
+# Import Pathing Guidelines for Spec files
+
+## The Problem
+One of the difficult problems that we deal with as library authors is defining the API that we expose to consuming application developers. This essentially means one of two things:
+  
+  1. We provide all the library exports that we use internally to consumers
+    * This is often a very bad idea for long term maintenance
+  2. We provide only a subset of the libraries' exports.
+    *  We can allow our consumers to use the parts of the library we intend them to use, leading to a smaller API surface.
+
+Unfortunately, we (the authors of Daffodil) often view the world from the first perspective. We often write our internal tests depending upon `imports` that are only available to us as library authors. 
+
+This means that the tests are from a very specific perspective, our internal view. So tests that pass for us won't necessarily pass for our consumers, as they may not have a necessary import available. Subsequently, this also means that code that works for us for example, may not work for consuming applications. 
```

Comment:
Hmm...I don't know how else to say this. I really don't like lines 11-13. It feels like they're dancing around what they're trying to say instead of just saying it. And not the good kind of dancing. The bad kind.